modules/transport-grpc/src/main/java/org/opensearch/transport/grpc/proto/request/search/query/BoolQueryBuilderProtoUtils.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.transport.grpc.proto.request.search.query;
+
+import org.opensearch.index.query.AbstractQueryBuilder;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.protobufs.BoolQuery;
+import org.opensearch.protobufs.MinimumShouldMatch;
+import org.opensearch.protobufs.QueryContainer;
+import org.opensearch.transport.grpc.spi.QueryBuilderProtoConverterRegistry;
+
+import java.util.List;
+
+/**
+ * Utility class for converting BoolQuery Protocol Buffers to OpenSearch objects.
+ * This class provides methods to transform Protocol Buffer representations of bool queries
+ * into their corresponding OpenSearch BoolQueryBuilder implementations for search operations.
+ */
+class BoolQueryBuilderProtoUtils {
+
+    // Registry for query conversion - injected by the gRPC plugin
+    private static QueryBuilderProtoConverterRegistry REGISTRY;
```

Comment:
Given this is only used from the converter, I'm thinking we can have the converter pass the registry directly to `fromProto` here and avoid storing/setting the static reference twice.